### DISCUSSION TITLE
Is WinUI the wrong approach for complex programs?

### DISCUSSION THREAD

**Discussion by amazatis** [microsoft/microsoft-ui-xaml#7840] [Q&A]
We have an Enterprise-Resource-Planning program with PlugIn concept in WPF. We want to rewrite the project completely in WinUI, but we already reach the limits with the PlugIn concept. A dynamic loading of DLLs works, but not the display of pages from the dynamic DLL in the main program (See: (https://github.com/microsoft/microsoft-ui-xaml/issues/6920) 
**Is WinUI the wrong approach for such complex programs?**

**Comment by MEK3DK** [comment]
Assuming you are talking about WinUI 3.
The short answer is yes, it is the wrong approach for now.
When you say comlex I also assume it is a large program and due to the amount of bugs and missing features WinUI 3 is not there yet. So you are stuck with WPF.

However depending on what Microsoft does (do they fix the bugs and continue working on WinUI 3), then in 1-2-3 years (depending on the needs of your program), you might be ready to start using WinUI 3.
